static_frame/core/sort_interface_mixin.py

```diff
@@ -0,0 +1,35 @@
+import typing_extensions as tp
+
+from static_frame.core.util import SortStatus, TNDArrayIntDefault
+
+
+class SortInterfaceMixin:
+    """Defines the interface for objects that can be sorted"""
+
+    __slots__ = ()
+
+    def __copy__(self) -> tp.Self:
+        """
+        Return a shallow copy of this container, with no data copied.
+        """
+        raise NotImplementedError()  # pragma: no cover
+
+    def _reverse(self, axis: int = 0) -> tp.Self:
+        """
+        Return a reversed copy of this container, with no data copied.
+        """
+        raise NotImplementedError()  # pragma: no cover
+
+    def _apply_ordering(
+        self,
+        order: TNDArrayIntDefault,
+        sort_status: SortStatus,
+        axis: int = 0,
+    ) -> tp.Self:
+        """
+        Return a copy of this container with the specified ordering applied along the index of axis
+        """
+        raise NotImplementedError()  # pragma: no cover
```

Comment:
I think this should be a Protocol instead of a concrete class. As is, this will live in the MRO and incur some run-time cost. If this is a Protocol, I believe it will be removed entirely at runtime and incur zero cost.

Reply:
Per our discussion, I will remove the subclass and opt for just duck-typing with type-ignores. I'll add an issue to explore metaclass resolution in a different PR! This will hopefully resolve the long standing issue of allowing us to add `abc.Sized` in the class MRO!